src/ansys/fluent/core/quantity/quantity.py

```diff
@@ -100,14 +100,17 @@ def _arithmetic_precheck(self, __value, caller=None) -> str:
             SI unit string of new quantity.
         """
         # Cannot perform operations between quantities with incompatible dimensions
-        if isinstance(__value, Quantity) and (self.dimensions != __value.dimensions):
+        if (
+            isinstance(__value, Quantity)
+            and caller not in ["__mul__", "__truediv__"]
+            and (self.dimensions != __value.dimensions)
+        ):
```

Comment:
@acarvalh-work : 
I'm noting the above code along with associated code like this:
```
    def __truediv__(self, __value):
        new_units = self._arithmetic_precheck(__value, caller="__truediv__")
        if isinstance(__value, Quantity):
            temp_dimensions = [
                dim - __value.dimensions[idx] for idx, dim in enumerate(self.dimensions)
            ]
            new_si_value = self.si_value / __value.si_value
            new_dimensions = q.Dimensions(dimensions=temp_dimensions)
```
To clean this up, we shall:
- remove the pre-checks
- associate arithmetic dimensions operations with the `Dimensions` class itself. 
- in those operators, either return a `Dimensions` object or throw if the operand is invalid for the operation

We can do the above refactoring as a follow-up.